### PR TITLE
Move dynamodb lock into its own crate

### LINKED ISSUE #186: Move dynmodb lock into its own crate
# Description

Once we are running it in production, we should refactor it into a standalone code.

This is a very useful piece of code to be shared with the rest of the community. We should even blog about it if we have time to do that.

### PR DESCRIPTION
**Pull request by viirya**
# Description

Move dynamodb lock into its own crate.

# Related Issue(s)

closes #186 

# Documentation

### CODE REVIEW

**Comment by viirya** [comment]
Hmm, I also saw the CI clippy error when running `cargo clippy --features s3-rustls` on current main without this change:

```
error[E0252]: the name `tls` is defined multiple times
  --> /Users/viirya/.cargo/registry/src/github.com-1ecc6299db9ec823/rusoto_core-0.46.0/src/lib.rs:25:5
   |
23 | use hyper_rustls as tls;
   |     ------------------- previous import of the module `tls` here
24 | #[cfg(feature = "native-tls")]
25 | use hyper_tls as tls;
   |     ^^^^^^^^^^^^^^^^ `tls` reimported here
   |
   = note: `tls` must be defined only once in the type namespace of this module
help: you can use `as` to change the binding name of the import
   |
25 | use hyper_tls as other_tls;
   |     ~~~~~~~~~~~~~~~~~~~~~~

error[E0599]: no function or associated item named `new` found for struct `hyper_rustls::HttpsConnector` in the current scope
   --> /Users/viirya/.cargo/registry/src/github.com-1ecc6299db9ec823/rusoto_core-0.46.0/src/request.rs:224:41
    |
224 |         let connector = HttpsConnector::new();
    |                                         ^^^ function or associated item not found in `hyper_rustls::HttpsConnector<_>`

error[E0599]: no function or associated item named `new` found for struct `hyper_rustls::HttpsConnector` in the current scope
   --> /Users/viirya/.cargo/registry/src/github.com-1ecc6299db9ec823/rusoto_core-0.46.0/src/request.rs:235:41
    |
235 |         let connector = HttpsConnector::new();
    |                                         ^^^ function or associated item not found in `hyper_rustls::HttpsConnector<_>`
```

File: dynamodb_lock/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+[package]
+name = "dynamodb_lock"
+description = "Distributed lock backed by Dynamodb"
+version = "0.4.1"
+authors = ["Qingping Hou <qph@scribd.com>"]
```
Comment (houqp) [inline dynamodb_lock/Cargo.toml:5]: I didn't write this, @mosyp did :D  @mosyp could you help suggest a change for your preferred email here?

**Review by houqp — APPROVED** [review]
It's time to publish this awesome crate to crates.io!

**Comment by houqp** [comment]
ooops, looks like there are some conflicts :(

**Comment by viirya** [comment]
Thanks @houqp ! Let me resolve the conflicts.

**Review by houqp — APPROVED** [review]

**Pull request merged**